src/app/researcher/study/[studyId]/resubmit/[orgSlug]/form.tsx

```diff
@@ -0,0 +1,90 @@
+'use client'
+
+import { Button, Group, Stack } from '@mantine/core'
+import { upsertStudyAction } from '@/app/researcher/study/request/[orgSlug]/actions'
+import React from 'react'
+import { useForm } from '@mantine/form'
+import {
+    StudyProposalFormValues,
+    studyProposalApiSchema,
+} from '@/app/researcher/study/request/[orgSlug]/study-proposal-form-schema'
+import { UploadStudyJobCode } from '@/app/researcher/study/request/[orgSlug]/upload-study-job-code'
+import { useRouter } from 'next/navigation'
+import { notifications } from '@mantine/notifications'
+import { SelectedStudy } from '@/server/actions/study.actions'
+import { useMutation } from '@tanstack/react-query'
+import { CancelButton } from '@/components/cancel-button'
+import { uploadFile } from '../../../request/[orgSlug]/study-proposal'
+
+export function ResubmitStudyCodeForm(props: { study: SelectedStudy }) {
+    const { study } = props
+    const router = useRouter()
+
+    const studyProposalForm = useForm<StudyProposalFormValues>({
+        initialValues: {
+            title: study.title,
+            piName: study.piName,
+            descriptionDocument: null,
+            irbDocument: null,
+            agreementDocument: null,
+            mainCodeFile: null,
+            additionalCodeFiles: [],
+        },
+    })
+
+    const { isPending, mutate: resubmitStudy } = useMutation({
+        mutationFn: async (formValues: StudyProposalFormValues) => {
+            if (!formValues.mainCodeFile) {
+                throw new Error('Please upload a main code file to resubmit.')
+            }
+
+            const studyInfo = studyProposalApiSchema.parse(formValues)
+
+            const { urlForMainCodeUpload, urlForAdditionalCodeUpload } = await upsertStudyAction({
+                studyId: study.id,
+                orgSlug: study.orgSlug,
+                studyInfo,
+                mainCodeFileName: formValues.mainCodeFile.name,
+                codeFileNames: formValues.additionalCodeFiles.map((f) => f.name),
+            })
+
```

Comment:
we don't want to update the study.  We only want to add a new job to it, so maybe this becomes `addJobToStudy` or something like that